src/interactive/AccordionPanelGroup.jsx

```diff
@@ -28,30 +30,62 @@ class AccordionPanelGroup extends React.Component {
 	 * Keeps track of the clicked panel in order to support AccordionPanelGroups that only
 	 * have one panel open at a time.
 	 */
-	setClickedPanel(clickedPanel) {
-		if (!this.props.multiSelectable) {
-			this.setState({ clickedPanel });
+	setOpenPanels(clickedPanel, isOpen) {
+
+		const isSaved = this.isPanelInState(clickedPanel);
+
+		// tell parent to store it
+		let panelsToSave;
+
+		if (isOpen && !isSaved) {
+			if (this.props.multiSelectable) {
+				const openPanels = [...this.state.openPanels];
+				openPanels.push(clickedPanel);
+				panelsToSave = openPanels;
+			} else {
+				panelsToSave = [clickedPanel];
+			}
+		} else if (!isOpen && isSaved) {
+			const openPanels = [...this.state.openPanels];
+			const filteredPanels = openPanels.filter((panel) => panel.props.clickId !== clickedPanel.props.clickId);
+			panelsToSave = filteredPanels;
 		}
+		this.setState({ openPanels: panelsToSave });
+	}
+
+	isPanelInState(panel) {
+		return this.state.openPanels.filter((openPanel) => openPanel.props.clickId === panel.props.clickId).length > 0;
 	}
 
 	/**
 	 * @param {Object} accordionPanel - `AccordionPanel` components to clone
-	 * @param {number} i - index of the `AccordionPanel`
-	 * @param {boolean} isOpen - whether the `AccordionPanel` is open or not
+	 * @param {number} i - index of the `AccordionPanel` used as the key
 	 * @returns {Array} `AccordionPanel` components with props from `AccordionPanelGroup`
 	 */
-	clonePanel(accordionPanel, key, isOpen) {
-		const panelProps = {
-			key,
+	clonePanel(accordionPanel, i, arr, initialize) {
+		const isOpen = (initialize) ? accordionPanel.props.isOpen : this.isPanelInState(accordionPanel);
```

Comment:
this `initialize` arg might be awkward and I'm sure this can use some cleanup. What I'm trying to do is have the AccordionPanelGroup assign `clickId` props when cloning the panels at first, and only save them into state on initialization.